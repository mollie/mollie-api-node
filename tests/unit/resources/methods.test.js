import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

import Methods from 'resources/methods';
import Method from 'models/method';

import response from '../__stubs__/methods.json';

const mock = new MockAdapter(axios);

describe('methods', () => {
  let methods;
  beforeEach(() => {
    methods = new Methods(axios.create());
  });

  it('should have a resource name and model', () => {
    expect(Methods.resource).toBe('methods');
    expect(Methods.model).toBe(Method);
  });

  describe('.get()', () => {
    const methodId = 'ideal';
    const error = { error: { message: 'The method id is invalid' } };

    mock.onGet(`/methods/${methodId}`).reply(200, response._embedded.methods[0]);
    mock.onGet('/methods/foo').reply(500, error);

    it('should return a method instance', () =>
      methods.get(methodId).then(result => {
        expect(result).toBeInstanceOf(Method);
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      methods.get(methodId, (err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Method);
        expect(result).toMatchSnapshot();
        done();
      });
    });

    it('should return an error for non-existing IDs', () =>
      methods
        .get('foo')
        .then(() => {
          throw new Error('Should reject');
        })
        .catch(err => {
          expect(err).toBe(error);
        }));

    it('should return an error with a callback for non-existing IDs', done => {
      methods.get('foo', (err, result) => {
        expect(err).toBe(error);
        expect(result).toBeUndefined();
        done();
      });
    });
  });

  describe('.all()', () => {
    mock.onGet('/methods').reply(200, response);

    it('should return a list of all methods', () =>
      methods.all().then(result => {
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
      }));

    it('should work with a callback', done => {
      methods.all((err, result) => {
        expect(err).toBeNull();
        expect(result).toBeInstanceOf(Array);
        expect(result).toHaveProperty('links');
        expect(result).toMatchSnapshot();
        done();
      });
    });
  });
});
