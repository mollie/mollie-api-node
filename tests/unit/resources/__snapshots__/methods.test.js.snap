// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`methods .all() should return a list of all methods 1`] = `
Array [
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/ideal",
        "type": "application/hal+json",
      },
    },
    "description": "iDEAL",
    "id": "ideal",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/creditcard",
        "type": "application/hal+json",
      },
    },
    "description": "Credit card",
    "id": "creditcard",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/creditcard.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/creditcard%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/banktransfer",
        "type": "application/hal+json",
      },
    },
    "description": "Bank transfer",
    "id": "banktransfer",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/banktransfer.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/banktransfer%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/bitcoin",
        "type": "application/hal+json",
      },
    },
    "description": "Bitcoin",
    "id": "bitcoin",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/bitcoin.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/bitcoin%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/inghomepay",
        "type": "application/hal+json",
      },
    },
    "description": "ING Home'Pay",
    "id": "inghomepay",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/inghomepay.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/inghomepay%402x.png",
    },
    "resource": "method",
  },
]
`;

exports[`methods .all() should work with a callback 1`] = `
Array [
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/ideal",
        "type": "application/hal+json",
      },
    },
    "description": "iDEAL",
    "id": "ideal",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/creditcard",
        "type": "application/hal+json",
      },
    },
    "description": "Credit card",
    "id": "creditcard",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/creditcard.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/creditcard%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/banktransfer",
        "type": "application/hal+json",
      },
    },
    "description": "Bank transfer",
    "id": "banktransfer",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/banktransfer.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/banktransfer%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/bitcoin",
        "type": "application/hal+json",
      },
    },
    "description": "Bitcoin",
    "id": "bitcoin",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/bitcoin.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/bitcoin%402x.png",
    },
    "resource": "method",
  },
  Method {
    "_links": Object {
      "self": Object {
        "href": "https://api.mollie.com/v2/methods/inghomepay",
        "type": "application/hal+json",
      },
    },
    "description": "ING Home'Pay",
    "id": "inghomepay",
    "image": Object {
      "size1x": "https://www.mollie.com/images/payscreen/methods/inghomepay.png",
      "size2x": "https://www.mollie.com/images/payscreen/methods/inghomepay%402x.png",
    },
    "resource": "method",
  },
]
`;

exports[`methods .get() should return a method instance 1`] = `
Method {
  "_links": Object {
    "self": Object {
      "href": "https://api.mollie.com/v2/methods/ideal",
      "type": "application/hal+json",
    },
  },
  "description": "iDEAL",
  "id": "ideal",
  "image": Object {
    "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
    "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
  },
  "resource": "method",
}
`;

exports[`methods .get() should work with a callback 1`] = `
Method {
  "_links": Object {
    "self": Object {
      "href": "https://api.mollie.com/v2/methods/ideal",
      "type": "application/hal+json",
    },
  },
  "description": "iDEAL",
  "id": "ideal",
  "image": Object {
    "size1x": "https://www.mollie.com/images/payscreen/methods/ideal.png",
    "size2x": "https://www.mollie.com/images/payscreen/methods/ideal%402x.png",
  },
  "resource": "method",
}
`;
